va/va_prot.h

```diff
@@ -0,0 +1,456 @@
+ * attribs[1].type = VAConfigAttribProtectedContentCipherBlockSize;
+ * attribs[2].type = VAConfigAttribProtectedContentCipherMode;
+ * attribs[3].type = VAConfigAttribProtectedContentCipherSampleType;
+ * attribs[4].type = VAConfigAttribProtectedContentUsage;
+ * vaGetConfigAttributes(va_dpy, VAProfileProtected,
+ *     VAEntrypointProtectedContent, attribs, 5);
+ * if ((attribs[1].value & VA_PC_CIPHER_AES) == 0) {
+ *     // not find desired cipher algorithm
+ *     assert(0);
+ * }
+ * if ((attribs[2].value & VA_PC_BLOCK_SIZE_128) == 0) {
+ *     // not find desired block size
+ *     assert(0);
+ * }
+ * if ((attribs[3].value & VA_PC_CIPHER_MODE_CBC) == 0) {
+ *     // not find desired counter mode
+ *     assert(0);
+ * }
+ * if ((attribs[4].value & VA_PC_SAMPLE_TYPE_SUBSAMPLE) == 0) {
+ *     // not find desired sample type
+ *     assert(0);
+ * }
+ * if ((attribs[5].value & VA_PC_USAGE_WIDEVINE) == 0) {
+ *     // not find desired usage
+ *     assert(0);
+ * }
+ * \endcode
+ *
+ * \section api_pc_setup Set up a protected content session
+ *
+ * The protected content session can be attached to VA decode/encode/vp context
+ * to do decryption/protection in the pipeline.
+ * Before creating a protected content session, it needs to create a config
+ * first via vaCreateConfig(). Then using this config id to create a protected
+ * content session via vaCreateProtectedSession().
+ *
+ * The general control flow is demonstrated by the following pseudo-code:
+ * \code
+ * // Create config
+ * VAConfigID config_id;
+ *
+ * attribs[0].value = VA_PC_CIPHER_AES;
+ * attribs[1].value = VA_PC_BLOCK_SIZE_128;
+ * attribs[2].value = VA_PC_CIPHER_MODE_CBC;
+ * attribs[3].value = VA_PC_SAMPLE_TYPE_SUBSAMPLE;
+ * attribs[4].value = VA_PC_USAGE_WIDEVINE;
+ * va_status = vaCreateConfig(va_dpy, VAProfileProtected,
+ *     VAEntrypointProtectedContent, attribs, 5, &config_id);
+ * CHECK_VASTATUS(va_status, "vaCreateConfig");
+ * \endcode
+ *
+ * Once the config is set up, we can create protected content session via
+ vaCreateProtectedSession().
+ * \code
+ * // Create a protected session
+ * VAProtectedSessionID crypto_session;
+ *
+ * va_status = vaCreateProtectedSession(va_dpy, config_id, &crypto_session);
+ * CHECK_VASTATUS(va_status, "vaCreateProtectedSession");
+ * \endcode
+ *
+ * \section api_pc_exec TEE communication via vaProtectedSessionExecute()
+ *
+ * Before start decryption/encryption, we may need to communicate with TEE to
+ * get encrypted decryption. We need to call vaProtectedSessionExecute() to
+ * get it. The following pseudo-code demonstrates getting session id via
+ * vaProtectedSessionExecute() as an example.
+ *
+ * \code
+ * uint32_t app_id = 0xFF;
+ * VABufferID buffer;
+ * VAProtectedSessionExecuteBuffer exec_buff = {0};
+ *
+ * exec_buff.function_id = GET_SESSION_ID
+ * exec_buff.input.data = nullptr;
+ * exec_buff.input.data_size = 0;
+ * exec_buff.output.data = &app_id;
+ * exec_buff.output.max_data_size = (uint32_t);
```

Comment:
`max_data_size = (uint32_t)` does not make sense. Can you, please, use:
```
exec_buff.output.max_data_size = sizeof(app_id);
```

Reply:
Fixed